### PR TITLE
chore(mojaloop/#3521): nodejs upgrade

### PR DESCRIPTION
**Pull request by oderayi**
chore(mojaloop/#3521): NodeJS version upgrade - https://github.com/mojaloop/project/issues/3521
- Upgraded nodejs from v16 to v18 LTS
- Applied CI changes as previously implemented in [sdk-scheme-adapter](https://github.com/mojaloop/sdk-scheme-adapter/pull/453)
- Updated NPM dependencies
- Re-implement husky integration due to API changes in husky's latest version
- Resolved all audit issues
- Fixed docker-compose issues
- Migrated `master` branch to `main` branch

### CODE REVIEW

File: audit-ci.jsonc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  // Only use one of ["low": true, "moderate": true, "high": true, "critical": true]
+  "moderate": true,
+  "allowlist": [
+   "GHSA-v88g-cgmw-v5xw",
+   "GHSA-phwq-j96m-2c2q",
+   "GHSA-282f-qqgm-c34q",
+   "GHSA-6vfc-qv3f-vr6c",
+   "GHSA-mjxr-4v3x-q3m4",
+   "GHSA-rjqq-98f6-6j3r",
+   "GHSA-p9pc-299p-vxgp"
```
Comment (vijayg10) [inline audit-ci.jsonc:13]: Can we add some comments here mentioned why we want to allow these?

**Review by vijayg10 — COMMENTED** [review]
Just added a comment, otherwise LGTM

File: audit-ci.jsonc (hunk 1 above)
Comment (oderayi) [reply audit-ci.jsonc]: @vijayg10 I added links to the advisory information for all the issues.

Comment (oderayi) [reply audit-ci.jsonc]: @vijayg10 I added links to the advisory for each issue.

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**